GitUI/CommandsDialogs/SettingsDialog/SettingsPageBase.cs

```diff
@@ -48,7 +48,7 @@ protected ISettingsPageHost PageHost
 
         public virtual SettingsPageReference PageReference => new SettingsPageReferenceByType(GetType());
 
-        protected internal IServiceProvider ServiceProvider { get; }
+        protected internal new IServiceProvider ServiceProvider { get; }
```

Comment:
Why do we need to shadow (i.e., `new`) here?

Reply:
I was not sure and did not try whether the settings pages would directly work with the new inherited property `GitExtensionsControl.ServiceProvider`, i.e. whether it becomes available early enough.
I have tried to declare the property `virtual` but this has not worked due to `protected internal`. (IMHO there are too many variations of visibility modifiers in C#.)

Reply:
Are you able to check if the change in this file can be reverted? And if not, I'd like to be better understand why we need it (and have that documented).
Other than that, :shipit:

Reply:
> Are you able to check if the change in this file can be reverted? And if not, I'd like to be better understand why we need it (and have that documented).

I have added the xmldoc:
```
        /// <summary>
        ///  Gets the instance of <see cref="IServiceProvider"/> as assigned in the constructor.
        ///  This property replaces the property <see cref="GitExtensionsControl.ServiceProvider"/>,
        ///  which would be available only for controls implementing <see cref="IGitModuleControl"/>
        ///  or for controls placed on a <see cref="IGitModuleForm"/>. Neither is the case for settings pages.
        /// </summary>
```

Alternatively, we could split `GitExtensionsControl` because it deals with two different things:
* translation
* hotkeys

ServiceProvider is only used for hotkeys (currently).
Settings pages do not use hotkeys. They just need translation.

The extraction of `TranslatedControl` seems to be no big deal. So, I am doing it in this PR.